### PR TITLE
Pass config ownership to pmemkv database and handle force_create option

### PR DESCRIPTION
**Pull request by igchor**
---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/336)

### CODE REVIEW

File: tests/engines/cmap_test.cc [hunk 1]
```diff
@@ -41,6 +41,8 @@ const std::string PATH = "/dev/shm/pmemkv";
 const size_t SIZE = 1024ull * 1024ull * 512ull;
 const size_t LARGE_SIZE = 1024ull * 1024ull * 1024ull * 2ull;
 
+static int count = 0;
```
Comment (marcinslusarz) [inline tests/engines/cmap_test.cc]: ?

**Comment by lukaszstolarczuk** [comment]
@szyrom, please make sure this PR is reviewed and merged as soon as possible (tomorrow morning); for bindings to update
@ldorau and @annamarcink - for your information - this should be the last PR with API changes before release

File: tests/engines/cmap_test.cc (hunk 1 above)
Comment (igchor) [reply tests/engines/cmap_test.cc]: Done

File: src/config.h [hunk 2]
```diff
@@ -77,7 +77,7 @@ struct config {
 		try {
 			std::string mkey(key);
 			std::vector<char> v((char *)&value,
-					    (char *)&value + sizeof(value));
+					    (char *)&value + sizeof(void *));
```
Comment (marcinslusarz) [inline src/config.h]: this doesn't fix anything and is actually backward - using variables in sizeof is preferred

File: src/config.h [hunk 3]
```diff
@@ -124,11 +124,11 @@ struct config {
 		void *ptr_ptr;
 
 		auto status = get(key, (const void **)&ptr_ptr, &size, nullptr);
-		if (status == status::OK && size != sizeof(value))
+		if (status == status::OK && size != sizeof(void *))
```
Comment (marcinslusarz) [inline src/config.h]: sizeof(ptr_ptr)

File: src/config.h [hunk 4]
```diff
@@ -124,11 +124,11 @@ struct config {
...
 			return status::CONFIG_TYPE_ERROR;
 
 		if (status == status::OK)
-			memcpy(value, ptr_ptr, sizeof(value));
+			memcpy(value, ptr_ptr, sizeof(void *));
```
Comment (marcinslusarz) [inline src/config.h]: sizeof(ptr_ptr)

**Pull request merged**